Fix typos.

Maikuolan opened the pull request.

szepeviktor commented:
@Maikuolan Try https://github.com/crate-ci/typos

Maikuolan commented:
> @Maikuolan Try https://github.com/crate-ci/typos

I'll take a look at that when I've got a moment. Cheers. :+1:

opr commented on plugins/woocommerce-blocks/assets/js/base/components/price-slider/index.tsx:
```diff
@@ -78,7 +78,7 @@ export interface PriceSliderProps {
 	step?: number;
 	/**
-	 * Wheter we're in the editor or not.
+	 * Whether we're in the editor or not.
```
While we're here, please can we remove the superfluous "or not"?

opr commented on plugins/woocommerce-blocks/assets/js/base/components/block-error-boundary/types.ts:
```diff
@@ -18,15 +18,15 @@ interface BlockErrorBase {
 	button?: React.ReactNode;
 	/**
-	 * Controls wether to show the error block or fail silently
+	 * Controls whether to show the error block or fail silently
```
```suggestion
	 * Controls whether to show the error block or fail silently.
```

opr reviewed (changes requested):
These changes look fine, thanks for taking care of them. I left a couple of suggestions that I think would be nice to change while we're tidying up.

opr commented:
@Maikuolan thanks! Did you try [crate-ci/typos](https://github.com/crate-ci/typos) ?

I think it would be good to see if it picks up anything in comments/docblocks but probably skip any of the code changes for this PR.

Maikuolan commented:
> Did you try [crate-ci/typos](https://github.com/crate-ci/typos) ?

Not yet. Plan to read through the documentation to figure out how to get it working today though. :-)

szepeviktor commented:
Let me help you.

```toml
# Typos works on the filesystem
[files]
extend-exclude = [
    ".git/",
    "any/file/or/directory/",
]
ignore-hidden = false

[default]
# Ignore things that are not an identifier (words together)
extend-ignore-re = [
    "[0-9A-Za-z/+]{76}",
    "\\bBig Sur\\b",
]

[default.extend-words]
# Lowercase words (parts of identifiers)
"contiguities" = "continuities"

[default.extend-identifiers]
# Keep these
"I_AM_AN_IDENTIFIER" = "I_AM_AN_IDENTIFIER"
# Correct these
"I_AM_A_TIPO" = "I_AM_A_TYPO"
```

Maikuolan commented:
Sorry for the delayed reply. Was planning to follow up last night, but had stuff come up, was working today, etc.. As happens.

Running `crate-ci/typos` as a simple GitHub action (loosely based on the examples provided at their documentation, I wrote a workflow to trigger on pull requests, then sent a pull request from the patch branch containing the new workflow back to the trunk at the same fork as a test), a few more legitimate typos have been revealed, though they seem to all be within various changelogs, and I'm apprehensive to touch any of those, as messing with historical changelogs, changelogs for already released versions, components and such, as an outside contributor, even if it's to just to fix typos, instinctively feels like a bit of a no-no and something which could be seen as in bad taste, so I'll leave those be (if a repository owner, core contributor or such wants to do that, no worries and all good, but that's different, and of course, their call, not mine).

A few weird false positives were also revealed, though I imagine said false positives could probably be avoided by some kind of exclusion rule when running the action or something (though I've only experimented very lightly thus far).

Said new workflow, of course, was written just for the sake of testing `crate-ci/typos` as an action against the WooCommerce repository, and hasn't been fine-tuned for the parent `woocommerce/woocommerce` repository (I imagine, integrating such a new workflow to `woocommerce/woocommerce`, and figuring out what fine-tuning it would actually entail, would require some discussion between those responsible for integrating new workflows regarding whether it would be preferable to run it for every new commit, for just pull requests, or whenever/however else, and things like that). I haven't fleshed out any custom configuration or such to the workflow in question, seeing as it's just as a test for now.

Anyway, the short of it: It seems to work well enough when it comes to finding typos, though any changes suggested by it should still likely best be reviewed by a human eye prior to merging.

The workflow run in question: https://github.com/Maikuolan/woocommerce/actions/runs/10199452289

Maikuolan commented:
Thinking on it, actually.. The commits which said workflow run ran against didn't include any of the commits in *this* pull request, meaning it failed to catch any of the typos being addressed by this pull request. Whether that reflects the ability of `crate-ci/typos` to catch such typos or merely reflects that I hadn't yet added any definitions to the workflow or added any custom configuration.. no idea. It's my first time running it, after all. '^^ Probably, I need to tell it what I want it to catch, I guess.

If we need to add a significant number of definitions for it to be effective though (i.e., creating our own dictionary of typos and adding it to the workflow or its configuration), I imagine that might grow rather sizeable rather quickly, be a bit more bulky than desirable for something like a GitHub action workflow, and could serve as an argument against integrating it as a workflow at all (though, they provide means other than just workflows for using `crate-ci/typos`, such an offline shell utility for checking typos and such, so.. not a complete deal breaker necessarily).

opr commented:
@Maikuolan thanks for looking into this in so much detail! I think you're right that changing typos in changelog files is not a good idea, moreover they're stored on WordPress.org in SVN, so updating it here would only fix it in one place, imo it's not worth doing.

I also think adding a workflow increases the scope of this PR too much, but I do appreciate you trying that out.

What I had in mind was just to run that script locally in your CLI or something, and just check if there are any other similar typos to the ones you just fixed (ignoring false positives, changelog files, and things you're not 100% sure on).

What do you think about doing that and committing any fixes you make?

Thanks again for your effort with the workflow! 🙇🏼

Maikuolan commented:
> What do you think about doing that and committing any fixes you make?

I can do that. :-)

Maikuolan commented:
Running it in CLI, it seems to be finding *much* more than it did with workflows. I'll commit as I fix.

Maikuolan commented:
Some of the typos found are in the names of hooks and functions; I'll leave those alone for the moment, until I can be reasonably sure that fixing those particular typos won't break anything. For now, aiming to stick with low-risk fixes (typos in comments, documentation, etc).

szepeviktor commented:
You could publish a list of know typos in `[default.extend-identifiers]`

Maikuolan commented:
Feel free to review/rebase/merge early if it starts looking like it's starting to get a bit big (e.g., increasing risk of imminent merge conflicts due to the number of affected files and such). I'll keep working at it, but if I find more after the PR has already been merged/closed, I can always just open a new PR for any newly found typos; no worries.

opr commented on docs/docs-manifest.json:
Let's undo this change, I believe the file is auto generated

opr commented on packages/js/expression-evaluation/package.json:
I think this is fine, the changed won't show up on npm until the package is re-released, and I don't think we should release a new version just to fix a typo in metadata. It'll be applied next time we make a change.

opr commented on plugins/woocommerce/src/Blocks/Templates/AbstractTemplateCompatibility.php:
This looks pretty innocuous but I'd prefer to skip _any_ code changes in this PR 🙇🏼

opr commented on plugins/woocommerce/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationController.php:
Same here, I'd prefer any code changes to be left out of this PR, as low risk as this seems.

opr commented on plugins/woocommerce/src/Internal/Admin/Coupons.php:
Same here for code changes, please could we skip it?

opr reviewed (changes requested):
Hi, thanks for all of your work! This looks pretty good now, I just ask that we remove any code changes. Good point on getting this merged before too many conflicts arise!

opr commented:
A second thought as well @Maikuolan - since there are a lot of lint errors on changed `.md` files, maybe we could extract those changes into a separate PR and address the lint issues on them at the same time. Doing the typo updates across the codebase _and_ lint changes to MD files in one PR seems like a bit too much. Do you agree?

opr commented:
@Maikuolan thanks for addressing those comments! Regarding the one in [`packages/js/expression-evaluation/package.json`](https://github.com/woocommerce/woocommerce/pull/50047/files/7565105da774c29d3a09b257d00e531415a8f5b0#diff-6c35f7751e75fbed0367fdc8a7b10f72f8d9154ad37661ee46341f2628b5e866) I actually thought we would keep that change, but just not expect it to reflect on npm until any other changes are made.

> So, split the changes done to markdown files to their own PR, and keep this PR for the rest?

Yes, I think that makes most sense, fixing all the errors here is going to be too much work.

Thanks again for the changes so far!

Maikuolan committed:
Fix typos.

Maikuolan commented:
> > So, split the changes done to markdown files to their own PR, and keep this PR for the rest?
> 
> Yes, I think that makes most sense, fixing all the errors here is going to be too much work.

Done. :-)

All markdown file changes can now be found at #50282, while all the rest remain here.

> Thanks again for the changes so far!

Happy to help. :-)

shaybanshee reviewed (approved):
All text changes are correct; base branch needs to be updated and checks need to pass before merging. @opr @Maikuolan Still seeing a ton of lint issues on this?

Maikuolan commented:
```
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php: FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php:  178 | WARNING | Comment refers to a TODO task "remove these assignments when refactoring segmenter classes to use query objects" (Generic.Commenting.Todo.TaskFound)
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
```

(Plus the same repeated again a few times for some other lines at the same workflow run). I *suspect*, those *particular* warnings are appearing because some of the typos currently fixed at this PR are typos belonging to `@todo` comments. E.g,

```diff
-       // @todo Remove these assignements when refactoring segmenter classes to use query objects.
+       // @todo Remove these assignments when refactoring segmenter classes to use query objects.
```

Though, the *other* lint errors I see being produced by the workflows:

```
Error: Project packages/js/date is being changed, but no change file in packages/js/date/packages/js/date/changelog/ is touched!

Use `pnpm --filter=./packages/js/date changelog add` to add a change file.
```

(Plus similar repeated again a few times at the same workflow run). I'm not sure exactly how to interpret that. Is it because some of the typos being fixed affect JS files, maybe? I'm not sure.

Maikuolan commented:
How're we doing with this?

opr commented:
@Maikuolan thanks for your patience.

> I'm not sure exactly how to interpret that. Is it because some of the typos being fixed affect JS files, maybe? I'm not sure.

This JS one would probably need a changelog file indeed, since we're updating code in a package, but you can skip adding an actual changelog entry by choosing "patch" and "update" at the prompts. Would you please add one for the packages that were changed by this PR?

> Use `pnpm --filter=./packages/js/date changelog add` to add a change file.

<img width="693" alt="image" src="https://github.com/user-attachments/assets/0bdc564f-8f6d-4a77-bd82-2ad20fea63e7">

As for the PHP warnings, I've asked for more advice on what we should do here, hope to have an update and some advice soon.

Maikuolan committed:
Merge trunk into patch-1 and resolve conflicts.

Maikuolan commented:
Latest commit (`cea98fd`) syncs up a bit closer with the trunk and resolves some merge conflicts that've emerged since the past day or so (at the time of this reply, there shouldn't be any conflicts anymore; whether that holds until this reply is seen by others, time will tell).

Having a few issues with my local copy of NPM right now (planning to sort that out later today), so haven't yet added the changelog. Will do so as soon as I've resolved said local NPM issues and reply again once done to confirm.

opr commented:
Thanks! In terms of the lint warnings, I think we're leaning toward merging with them present. Ignoring them seems like a poor solution as they should be addressed.

Manussakis commented:
Hey @Maikuolan !

I noticed you fixed typos on files inside the `plugins/woocommerce/src/Admin/Features/ProductBlockEditor` directory.

Maybe you want to fix one last typo on the `ProductFormsController.php` file there.

**ot**

```php
/**
 * Create ot update a product_form post for each product form template."
```

Thanks!

Maikuolan committed:
Merge trunk into patch-1.

Favouring trunk where conflicts occur; Easier to drop my own changes for
the affected files due to how numerous the changes are at trunk for the
affected files. Can refix typos for the affected files again later if they
persist.

Maikuolan committed:
Fix typo (#issuecomment-2304288161).

Maikuolan committed:
Refix (some) typos.

Won't bother with the to-do fixes, as it'll just mean more problems with
some of the tests again.

Maikuolan commented:
To-do comment typo fixes are gone now (had to resolve conflicts again, some which affected those files; easier to just drop those particular fixes). Conflicts now resolved again a trunk up to-date (again).

> Maybe you want to fix one last typo on the `ProductFormsController.php` file there.
> 
> **ot**

Done. :-)

NPM still giving me grief at my machine (OS dropped paths for whatever reason and some other problems that you guys don't need to hear my whinge about as not important/relevant for the PR; didn't get time to look into it further today due to work, but planning to look into it again after I post this reply comment). Anyway, will sort out what's needed once able.

Maikuolan commented:
Would it be okay for me to try to just create the changelogs manually? When trying to do it via NPM, I'm seeing this (and I'm not really sure how to address NPM's complaints; `pnpm i -g pnpm` doesn't seem to do anything at all):

```
$ pnpm --filter=./packages/js/date changelog add
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your pnpm version is incompatible with "D:\Git\Repo\woocommerce".

Expected version: 9.1.3
Got: 9.8.0

This is happening because the package's manifest has an engines.pnpm field specified.
To fix this issue, install the required pnpm version globally.

To install the latest version of pnpm, run "pnpm i -g pnpm".
To check your pnpm version, run "pnpm -v".
```

Maikuolan committed:
Manually add some changelogs for #50047.

Maikuolan commented:
Latest commit (`e480de3`) adds changelogs for:
- packages/js/date
- packages/js/number
- plugins/woo-ai
- plugins/woocommerce-admin
- plugins/woocommerce

That should cover all the packages which own files affected by this pull request, except for.. plugins/woocommerce-blocks and packages/js/expression-evaluation.

As for plugins/woocommerce-blocks: I don't see any changelogs directory provided for it, so I'm not sure what would be the best location for such a changelog, or whether it might be better to just leave it alone anyway. Let me know if it needs one please (and where it should go).

As for packages/js/expression-evaluation: The only affected file owned by that package is its package.json file, and as per the review comment at #discussion_r1701578104, I assume a changelog for that one typo won't be necessary right now (but let me know if that assumption is wrong, as it's easy and simple enough to do the same for that as has been done for the others).

These changelogs have been added manually via my editor (NPM at my local machine still not playing nice). That being the case, please let me know if I've gotten anything wrong, used the wrong format for the changelogs, whether there's something else that needs to be done that I might've missed (e.g., maybe NPM doing something else in the background regarding changelogs that I hadn't accounted for with that latest commit), and also more broadly whether this pull request is still missing anything in regard to successfully passing all tests/workflows.

Maikuolan committed:
Merge trunk into patch-1.

opr commented:
Hi @Maikuolan thanks for adding the changelogs here - manually is fine. To get around your pnpm error, what you might try is:

`corepack enable && corepack prepare pnpm@9.1.3 --activate`

This works for me, but since you added them manually I suppose this isn't super important anymore!

Noticed that there's still one missing from `packages/js/expression-evaluation/packages/js/expression-evaluation/changelog/ ` if you could take care of that I would appreciate it.

As for the WC Blocks one, I'm not certain we _need_ one there, it should have been reported in the lint job if so, let's fix the one above and check.

Thanks!

Maikuolan committed:
Manually add a changelog for #50047 (packages/js/expression-evaluation).

Maikuolan committed:
Merge trunk into patch-1.

Maikuolan commented:
Done. :-)

opr commented:
Thanks a lot! Looks good on changelog use for now 🤞🏼 for the tests etc.

Maikuolan commented:
`CI / Core e2e tests 4/6`:
```
Error: Error establishing a database connection. This either means that the username and password information in your `wp-config.php` file is incorrect or that contact with the database server at `tests-mysql` could not be established. This could mean your host’s database server is down.
```
*Appears* like a possible timeout to me (so, could be a temporary problem, due to too many other tests running at the same time; I'm just speculating there, though). Maybe we could try re-run that specific test..?

`Build Live Branch / build (pull_request)`:
```
Error: Unhandled error: HttpError: Resource not accessible by integration
```
That one, I have no idea. o.0

Maikuolan commented:
Nice. `CI / Core e2e tests 4/6` passing now. :-)

opr reviewed (approved):
@Maikuolan thanks a lot for sticking with this while we figure out all the bumps. Really appreciate your time and patience here. Let's merge this! 🥳
